README.md

```diff
@@ -0,0 +1,42 @@
+# Code Test
+
+## Decisions
+
+### No frameworks
+I have been work with some Go web "frameworks" like Gorilla Mux and Gin.  
+But for this test I decided to not use them just to play with the guts of http package.  
+Then, I learned to value them because although they are really simple they help a lot.  
+
+### Loosely coupling
+The Warehouse ElementGetter interface make it easy to change beetween data sources.  
+For now, it is just csv, but anything else that implements the GetElements function will be  
+possible to use as data source.  
+
+Also, the csv service receives a list of validators so we can add more validations to it without change the csv service.  
+
+## The Challenge
+What nice problem to solve.  
+I started thinking that a linked list would solve the problem.  
```

Comment:
Really appreciate hearing how you thought about the problem. A set of branches and draft PRs would have been nicer to see these experiments.